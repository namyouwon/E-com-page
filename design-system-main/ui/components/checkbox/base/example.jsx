// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';
import _ from '../../../shared/helpers';
import {
  FormElement,
  FormElementControl,
  FormElementTooltip
} from '../../form-element';
const checkboxLabel = 'Checkbox Label';
const errorId = 'error_01';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

let Fieldset = props => (
  <fieldset
    className={classNames(
      'slds-form-element',
      { 'slds-is-required': props.required },
      props.className
    )}
  >
    {props.children}
  </fieldset>
);

let Legend = props => (
  <legend className="slds-form-element__legend slds-form-element__label">
    {props.children}
  </legend>
);

export let Checkbox = props => {
  const uniqueId = _.uniqueId('checkbox-unique-id-'); // "*-unique-id-*" prevents copy/paste & informs about IDs

  return (
    <div className={classNames('slds-checkbox', props.className)}>
      {props.children}
      <input
        type="checkbox"
        name={props.name || 'options'}
        id={props.id ? props.id : uniqueId}
        value={props.id ? props.id : uniqueId}
        disabled={props.disabled}
        defaultChecked={props.checked}
        aria-describedby={props.errorId}
        tabIndex={props.tabIndex}
        aria-labelledby={
          props.labelId && props.groupId
            ? props.labelId + ' ' + props.groupId
            : null
        }
        ref={el => el && (el.indeterminate = props.isIndeterminate)}
      />
      <label
        className={classNames('slds-checkbox__label')}
        htmlFor={props.id ? props.id : uniqueId}
        id={props.labelId}
      >
        <span className="slds-checkbox_faux" />
        {props.label && (
          <span
            className={classNames(
              'slds-form-element__label',
              props.hideLabel ? 'slds-assistive-text' : null
            )}
          >
            {props.isRequired && (
              <abbr className="slds-required" title="required">
                *{' '}
              </abbr>
            )}
            {props.label}
          </span>
        )}
      </label>
      {props.hasTooltip && (
        <FormElementTooltip showTooltip={props.showTooltip} />
      )}
    </div>
  );
};

/// ///////////////////////////////////////////
// State Constructor(s)
/// ///////////////////////////////////////////

export let Indeterminate = props => {
  const uniqueId = _.uniqueId('checkbox-unique-id-');

  return (
    <FormElement>
      <Checkbox
        id={uniqueId}
        label="Indeterminate Checkbox Label"
        isIndeterminate
      />
    </FormElement>
  );
};

export let Required = props => (
  <FormElement>
    <Checkbox label={checkboxLabel}>
      <abbr className="slds-required" title="required">
        *
      </abbr>
    </Checkbox>
  </FormElement>
);

export let ErrorState = props => (
  <FormElement
    hasError
    inlineMessage="This field is required"
    errorId={errorId}
  >
    <Checkbox label={checkboxLabel} errorId="error_01">
      <abbr className="slds-required" title="required">
        *
      </abbr>
    </Checkbox>
  </FormElement>
);

export let Disabled = props => (
  <FormElement>
    <Checkbox label={checkboxLabel} disabled />
  </FormElement>
);

export let CheckedDisabled = props => (
  <FormElement>
    <Checkbox label={checkboxLabel} checked disabled />
  </FormElement>
);

export let Group = props => (
  <Fieldset>
    <Legend>Checkbox Group Label</Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} checked />
      <Checkbox label={checkboxLabel} />
    </FormElementControl>
  </Fieldset>
);

export let GroupRequired = props => (
  <Fieldset required>
    <Legend>
      <abbr className="slds-required" title="required">
        *
      </abbr>{' '}
      Checkbox Group Label
    </Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} checked />
      <Checkbox label={checkboxLabel} />
    </FormElementControl>
  </Fieldset>
);

export let GroupError = props => (
  <Fieldset required className="slds-has-error">
    <Legend>
      <abbr className="slds-required" title="required">
        *
      </abbr>{' '}
      Checkbox Group Label
    </Legend>
    <FormElementControl>
      <Checkbox errorId={errorId} label={checkboxLabel} checked />
      <Checkbox errorId={errorId} label={checkboxLabel} />
    </FormElementControl>
    <div id={errorId} className="slds-form-element__help">
      This field is required
    </div>
  </Fieldset>
);

export let GroupDisabled = props => (
  <Fieldset>
    <Legend>Checkbox Group Label</Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} disabled />
      <Checkbox label={checkboxLabel} disabled />
    </FormElementControl>
  </Fieldset>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <FormElement>
        <Checkbox label={checkboxLabel} checked />
      </FormElement>
    )
  }
];

export let states = [
  {
    id: 'required',
    label: 'Required',
    element: <Required />
  },
  {
    id: 'required-with-long-label',
    label: 'Required with long label',
    demoStyles: 'width: 200px',
    storybookStyles: true,
    element: (
      <FormElement>
        <Checkbox label="When you wake up in the morning, it's a quarter to one, And you want to have a little fun. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch">
          <abbr className="slds-required" title="required">
            *
          </abbr>
        </Checkbox>
      </FormElement>
    )
  },
  {
    id: 'required-not-form-element-long-label',
    label: 'Required (Not Form Element) with long label',
    demoStyles: 'width: 200px',
    storybookStyles: true,
    element: (
      <Checkbox label="When you wake up in the morning, it's a quarter to one, And you want to have a little fun. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch">
        <abbr className="slds-required" title="required">
          *
        </abbr>
      </Checkbox>
    )
  },
  {
    id: 'required-optional-mix',
    label: 'Required/Optional Combination',
    element: (
      <FormElement>
        <Checkbox label={checkboxLabel}>
          <abbr className="slds-required" title="required">
            *
          </abbr>
        </Checkbox>
        <Checkbox label={checkboxLabel} />
      </FormElement>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: <ErrorState />
  },
  {
    id: 'disabled',
    label: 'Disabled',
    element: <Disabled />
  },
  {
    id: 'checked-and-disabled',
    label: 'Checked and Disabled',
    element: <CheckedDisabled />
  },
  {
    id: 'indeterminate',
    label: 'Indeterminate',
    element: <Indeterminate />
  }
];

export let examples = [
  {
    id: 'wrapping-text',
    label: 'Checkboxes with Wrapping Text',
    demoStyles: `width: 210px`,
    storybookStyles: true,
    element: (
      <FormElement>
        <Checkbox label="Early Morning (12 AM - 8 AM)" checked />
        <Checkbox label="Morning (8 AM - 12 PM)" />
        <Checkbox label="Afternoon (12 PM - 5 PM)" />
        <Checkbox label="Evening (5 PM - 12 AM)" />
      </FormElement>
    )
  },
  {
    id: 'wrapping-text-required-group',
    label: 'Required Checkboxes with Wrapping Text',
    demoStyles: `width: 210px`,
    storybookStyles: true,
    element: (
      <Fieldset>
        <Legend>
          <abbr className="slds-required" title="required">
            *{' '}
          </abbr>
          Checkbox Group Label
        </Legend>
        <FormElementControl>
          <Checkbox label="Early Morning (12 AM - 8 AM)" />
          <Checkbox label="Morning (8 AM - 12 PM)" />
          <Checkbox label="Afternoon (12 PM - 5 PM)" />
          <Checkbox label="Evening (5 PM - 12 AM)" />
        </FormElementControl>
      </Fieldset>
    )
  },
  {
    id: 'long-wrapping-text-required-group',
    label: 'Required Checkboxes with Abnormally-long Wrapping Text',
    demoStyles: `width: 210px`,
    storybookStyles: true,
    element: (
      <Fieldset>
        <Legend>
          <abbr className="slds-required" title="required">
            *{' '}
          </abbr>
          Checkbox Group Label
        </Legend>
        <FormElementControl>
          <Checkbox label="When you wake up in the morning, it's a quarter to one, And you want to have a little fun. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch" />
          <Checkbox label="When you wake up in the morning, it's a quarter to two, And you want to find, something to do. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch" />
          <Checkbox label="And when you wake up in the morning, it's a quarter to three, And your mind starts hummin', twiddle dee dee. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch" />
          <Checkbox label="When you wake up in the morning, it's a quarter to four, And you think you hear a knock on the door. You brush your teeth, ch ch ch ch, ch ch ch ch. You brush your teeth, ch ch ch ch, ch ch ch ch" />
        </FormElementControl>
      </Fieldset>
    )
  },
  {
    id: 'group',
    label: 'Group',
    element: <Group />
  },
  {
    id: 'group-required',
    label: 'Group Required',
    element: <GroupRequired />
  },
  {
    id: 'group-error',
    label: 'Group with error',
    element: <GroupError />
  },
  {
    id: 'group-disabled',
    label: 'Group Disabled',
    element: <GroupDisabled />
  },
  {
    id: 'stacked',
    label: 'Stacked',
    element: (
      <FormElement>
        <Checkbox
          className="slds-checkbox_stacked"
          label={checkboxLabel}
          checked
        />
      </FormElement>
    )
  },
  {
    id: 'rtl',
    label: 'Right to Left',
    element: (
      <div dir="rtl">
        <FormElement>
          <Checkbox label={checkboxLabel} checked />
        </FormElement>
      </div>
    )
  },
  {
    id: 'rtl-with-legend',
    label: 'Right to Left with legend',
    element: (
      <div dir="rtl">
        <Group />
      </div>
    )
  }
];
