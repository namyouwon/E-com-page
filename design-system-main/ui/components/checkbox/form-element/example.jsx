// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { FormElementCheckboxStandalone } from './';
import { FormElement } from '../../form-element/';
import { UtilityIcon } from '../../icons/base/example';

export default <FormElementCheckboxStandalone />;

export let states = [
  {
    id: 'checked',
    label: 'Checked',
    element: <FormElementCheckboxStandalone isChecked />
  },
  {
    id: 'disabled',
    label: 'Disabled',
    element: <FormElementCheckboxStandalone isDisabled />
  },
  {
    id: 'checked-disabled',
    label: 'Checked and Disabled',
    element: <FormElementCheckboxStandalone isChecked isDisabled />
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <FormElementCheckboxStandalone
        isRequired
        hasError
        inlineMessage="This field is required"
      />
    )
  },
  {
    id: 'required',
    label: 'Required',
    element: <FormElementCheckboxStandalone isRequired />
  },
  {
    id: 'view-mode-unchecked',
    label: 'View mode - Unchecked',
    element: (
      <FormElement labelContent="Form Element Label" isViewMode>
        <UtilityIcon
          symbol="steps"
          size="x-small"
          useCurrentColor
          assistiveText="False"
          title="False"
        />
      </FormElement>
    )
  },
  {
    id: 'view-mode-checked',
    label: 'View mode - Checked',
    element: (
      <FormElement labelContent="Form Element Label" isViewMode>
        <UtilityIcon
          symbol="check"
          size="x-small"
          useCurrentColor
          assistiveText="True"
          title="True"
        />
      </FormElement>
    )
  }
];

export let examples = [
  {
    id: 'help-text-icon',
    label: 'Help text icon',
    element: <FormElementCheckboxStandalone hasTooltip />
  },
  {
    id: 'required-help-text-icon',
    label: 'Required with Help text icon',
    element: <FormElementCheckboxStandalone isRequired hasTooltip />
  },
  {
    id: 'required-help-text-icon-tooltip',
    label: 'Required with Help text icon, showing tooltip',
    element: (
      <div style={{ paddingTop: '3rem' }}>
        <FormElementCheckboxStandalone isRequired hasTooltip showTooltip />
      </div>
    )
  }
];
