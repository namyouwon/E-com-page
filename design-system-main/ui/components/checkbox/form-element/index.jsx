// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import { CheckboxStandalone } from '../';
import { FormElement } from '../../form-element/';
import _ from '../../../shared/helpers';

// Component for standalone checkboxes within form
export const FormElementCheckboxStandalone = props => {
  const {
    name,
    value,
    isDisabled,
    isChecked,
    isRequired,
    isEditing,
    isStacked,
    isHorizontal,
    labelContent,
    hasError,
    hasTooltip,
    showTooltip,
    inlineMessage
  } = props;
  const uniqueId = _.uniqueId('checkbox-unique-id-'); // "*-unique-id-*" prevents copy/paste & informs about using unique IDs
  const uniqueErrorId = _.uniqueId('error-unique-id-');

  return (
    <FormElement
      labelContent={labelContent || 'Form Element Label'}
      inputId={uniqueId}
      isRequired={isRequired}
      isEditing={isEditing}
      isStacked={isStacked}
      isHorizontal={isHorizontal}
      hasError={hasError}
      inlineMessage={inlineMessage}
      errorId={hasError && uniqueErrorId}
      hasTooltip={hasTooltip}
      showTooltip={showTooltip}
    >
      <CheckboxStandalone
        id={uniqueId}
        name={name}
        value={value}
        isDisabled={isDisabled}
        isChecked={isChecked}
        isRequired={isRequired}
        errorId={hasError && uniqueErrorId}
      />
    </FormElement>
  );
};

CheckboxStandalone.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  isDisabled: PropTypes.bool,
  isChecked: PropTypes.bool,
  isRequired: PropTypes.bool,
  isEditing: PropTypes.bool,
  isStacked: PropTypes.bool,
  isHorizontal: PropTypes.bool,
  labelContent: PropTypes.string,
  hasError: PropTypes.bool,
  hasTooltip: PropTypes.bool,
  showTooltip: PropTypes.bool,
  inlineMessage: PropTypes.string
};
