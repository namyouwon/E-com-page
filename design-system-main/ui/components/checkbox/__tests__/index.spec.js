// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { Checkbox } from '../base/example';
import { CheckboxPrimitive, CheckboxStandalone } from '../';
import { FormElementCheckboxStandalone } from '../form-element';

const { matchesMarkup } = createHelpers(__dirname);

describe('Checkbox', () => {
  it('renders checkbox', () => matchesMarkup(<Checkbox />));
  it('renders checkbox with a label', () =>
    matchesMarkup(<Checkbox label="label" />));
  it('renders checkbox with a hidden label', () =>
    matchesMarkup(<Checkbox label="label" hideLabel />));
  it('renders checkbox with required asterisk', () =>
    matchesMarkup(<Checkbox isRequired />));
  it('renders checkbox that is disabled', () =>
    matchesMarkup(<Checkbox disabled />));
  it('renders checkbox with help text icon', () =>
    matchesMarkup(<Checkbox hasTooltip />));
});

describe('Checkbox Primitive', () => {
  it('renders primitive checkbox', () => matchesMarkup(<CheckboxPrimitive />));
  it('renders primitive checkbox with an ID', () =>
    matchesMarkup(<CheckboxPrimitive id="checkbox-id" />));
  it('renders primitive checkbox with a name', () =>
    matchesMarkup(<CheckboxPrimitive name="checkbox-name" />));
  it('renders primitive checkbox with a value', () =>
    matchesMarkup(<CheckboxPrimitive value="checkbox-value" />));
  it('renders primitive checkbox that is disabled', () =>
    matchesMarkup(<CheckboxPrimitive isDisabled />));
  it('renders primitive checkbox that is required', () =>
    matchesMarkup(<CheckboxPrimitive isRequired />));
  it('renders primitive checkbox with an aria-describedby error ID', () =>
    matchesMarkup(<CheckboxPrimitive errorId="error-id" />));
});

describe('Standalone Checkbox', () => {
  it('renders standalone checkbox', () =>
    matchesMarkup(<CheckboxStandalone />));
  it('renders primitive checkbox with an ID', () =>
    matchesMarkup(<CheckboxStandalone id="checkbox-id" />));
  it('renders primitive checkbox with a name', () =>
    matchesMarkup(<CheckboxStandalone name="checkbox-name" />));
  it('renders primitive checkbox with a value', () =>
    matchesMarkup(<CheckboxStandalone value="checkbox-value" />));
  it('renders primitive checkbox that is disabled', () =>
    matchesMarkup(<CheckboxStandalone isDisabled />));
  it('renders primitive checkbox that is required', () =>
    matchesMarkup(<CheckboxStandalone isRequired />));
  it('renders primitive checkbox with an aria-describedby error ID', () =>
    matchesMarkup(<CheckboxStandalone errorId="error-id" />));
});

describe('Form Element Standalone Checkbox', () => {
  it('renders form element standalone checkbox with a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone />));
  it('renders form element standalone checkbox with an ID and a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone id="checkbox-id" />));
  it('renders form element standalone checkbox with a name and a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone name="checkbox-name" />));
  it('renders form element standalone checkbox with a value and a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone value="checkbox-value" />));
  it('renders form element standalone checkbox that is disabled with a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone isDisabled />));
  it('renders form element standalone checkbox that is required with a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone isRequired />));
  it('renders = form element standalone checkbox with an aria-describedby error ID with label and error text', () =>
    matchesMarkup(
      <FormElementCheckboxStandalone hasError inlineMessage="Error message" />
    ));
  it('renders form element standalone checkbox that is checked with a label', () =>
    matchesMarkup(<FormElementCheckboxStandalone isChecked />));
});
