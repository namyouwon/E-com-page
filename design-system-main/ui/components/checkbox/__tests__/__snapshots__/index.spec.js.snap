// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Primitive renders primitive checkbox 1`] = `
<input
  name="label"
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox that is disabled 1`] = `
<input
  disabled={true}
  name="label"
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox that is required 1`] = `
<input
  name="label"
  required={true}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with a name 1`] = `
<input
  name="checkbox-name"
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with a value 1`] = `
<input
  name="label"
  type="checkbox"
  value="checkbox-value"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with an ID 1`] = `
<input
  id="checkbox-id"
  name="label"
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with an aria-describedby error ID 1`] = `
<input
  aria-describedby="error-id"
  name="label"
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox renders checkbox 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    id="checkbox-unique-id-1"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-1"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-1"
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Checkbox renders checkbox that is disabled 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    disabled={true}
    id="checkbox-unique-id-5"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-5"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-5"
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Checkbox renders checkbox with a hidden label 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    id="checkbox-unique-id-3"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-3"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-3"
  >
    <span
      className="slds-checkbox_faux"
    />
    <span
      className="slds-form-element__label slds-assistive-text"
    >
      label
    </span>
  </label>
</div>
`;

exports[`Checkbox renders checkbox with a label 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    id="checkbox-unique-id-2"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-2"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-2"
  >
    <span
      className="slds-checkbox_faux"
    />
    <span
      className="slds-form-element__label"
    >
      label
    </span>
  </label>
</div>
`;

exports[`Checkbox renders checkbox with help text icon 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    id="checkbox-unique-id-6"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-6"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-6"
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
  <div
    className="slds-form-element__icon"
  >
    <button
      className="slds-button slds-button_icon"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Help
      </span>
    </button>
  </div>
</div>
`;

exports[`Checkbox renders checkbox with required asterisk 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-labelledby={null}
    id="checkbox-unique-id-4"
    name="options"
    type="checkbox"
    value="checkbox-unique-id-4"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-unique-id-4"
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Form Element Standalone Checkbox renders = form element standalone checkbox with an aria-describedby error ID with label and error text 1`] = `
<div
  className="slds-form-element slds-has-error"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-13"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        aria-describedby="error-unique-id-7"
        id="checkbox-unique-id-13"
        name="label"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
  <div
    className="slds-form-element__help"
    id="error-unique-id-7"
  >
    Error message
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox that is checked with a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-14"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        defaultChecked={true}
        id="checkbox-unique-id-14"
        name="label"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox that is disabled with a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-11"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        disabled={true}
        id="checkbox-unique-id-11"
        name="label"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox that is required with a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-12"
  >
    <abbr
      className="slds-required"
      title="required"
    >
      *
       
    </abbr>
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        id="checkbox-unique-id-12"
        name="label"
        required={true}
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox with a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-7"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        id="checkbox-unique-id-7"
        name="label"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox with a name and a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-9"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        id="checkbox-unique-id-9"
        name="checkbox-name"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox with a value and a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-10"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        id="checkbox-unique-id-10"
        name="label"
        type="checkbox"
        value="checkbox-value"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Form Element Standalone Checkbox renders form element standalone checkbox with an ID and a label 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="checkbox-unique-id-8"
  >
    Form Element Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <span
      className="slds-checkbox slds-checkbox_standalone"
    >
      <input
        id="checkbox-unique-id-8"
        name="label"
        type="checkbox"
        value="option"
      />
      <span
        className="slds-checkbox_faux"
      />
    </span>
  </div>
</div>
`;

exports[`Standalone Checkbox renders primitive checkbox that is disabled 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    disabled={true}
    name="label"
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox that is required 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    name="label"
    required={true}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with a name 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    name="checkbox-name"
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with a value 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    name="label"
    type="checkbox"
    value="checkbox-value"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with an ID 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    id="checkbox-id"
    name="label"
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with an aria-describedby error ID 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby="error-id"
    name="label"
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders standalone checkbox 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    name="label"
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;
